### PR TITLE
lib/ffmpeg/encoderbase: use exe2os and filepath2os

### PR DESCRIPTION
**Pull request by tong1wu**
Ensure ffmpeg encoder command is compatible with Windows os.

Ensure input and output files are converted to native os
path format.

Signed-off-by: Tong Wu <tong1.wu@intel.com>

### CODE REVIEW

**Comment by tong1wu** [comment]
@Bin-CI Thanks for the suggestion, just modified the patch

**Comment by tong1wu** [comment]
> Miss `filepath2os` of `-y` option: https://github.com/intel/vaapi-fits/blob/master/lib/ffmpeg/encoderbase.py#L93 and https://github.com/intel/vaapi-fits/blob/master/lib/ffmpeg/encoderbase.py#L246

Thanks, already updated

File: lib/ffmpeg/encoderbase.py [hunk 1]
```diff
@@ -90,7 +90,7 @@ def gen_output_opts(self):
     if get_media()._get_gpu_gen() >= 11 and self.codec.startswith("hevc"):
       opts += " -b_strategy 1"
 
-    opts += " -vframes {frames} -y {encoded}"
+    opts += f" -vframes {self.frames} -y {filepath2os(self.encoded)}"
```
[the remark this answers is not in the document]
Comment (uartie) [reply lib/ffmpeg/encoderbase.py:93]: pre-check also fails for this reason.

[the remark this answers is not in the document]
Comment (tong1wu) [reply lib/ffmpeg/encoderbase.py]: Thanks, already modified as suggested.

File: lib/ffmpeg/encoderbase.py [hunk 2]
```diff
@@ -187,12 +188,15 @@ def encode(self):
 
     get_media().test_call_timeout = vars(self).get("call_timeout", 0)
 
-    iopts = self.gen_input_opts()
-    oopts = self.gen_output_opts()
     name  = self.gen_name().format(**vars(self))
     ext   = self.get_file_ext()
 
     self.encoded = get_media()._test_artifact("{}.{}".format(name, ext))
+    self.osencoded = filepath2os(self.encoded)
```
Comment (uartie) [inline lib/ffmpeg/encoderbase.py:195]: Also need to do this again after line 207 (i.e. after `self.encoded = get_media()._test_artifact("{}_{}.{}".format(name, i, ext))`) for r2r cases

**Review by uartie — CHANGES_REQUESTED** [review]

File: lib/ffmpeg/encoderbase.py (hunk 2 above)
Comment (tong1wu) [reply lib/ffmpeg/encoderbase.py:195]: Thanks, already modified

File: lib/ffmpeg/encoderbase.py [hunk 3]
```diff
@@ -241,11 +246,13 @@ def check_metrics(self):
       iopts += "-c:v {ffdecoder} "
     iopts += "-i {encoded}"
```
Comment (uartie) [inline lib/ffmpeg/encoderbase.py:247]: Do we need `"-i {osencoded}"` here too?

File: lib/ffmpeg/encoderbase.py [hunk 4]
```diff
@@ -261,7 +268,8 @@ def check_level(self):
       return
 
     output = call(
-      "ffprobe -i {encoded} -v quiet -show_entries stream=level"
+      f"{exe2os('ffprobe')}"
+      " -i {encoded} -v quiet -show_entries stream=level"
```
Comment (uartie) [inline lib/ffmpeg/encoderbase.py:272]: `" -i {osencoded}"`?

File: lib/ffmpeg/encoderbase.py [hunk 5]
```diff
@@ -270,7 +278,8 @@ def check_forced_idr(self):
       return
 
     output = call(
-      "ffmpeg -v verbose -i {encoded} -c:v copy"
+      f"{exe2os('ffmpeg')}"
+      " -v verbose -i {encoded} -c:v copy"
```
Comment (uartie) [inline lib/ffmpeg/encoderbase.py:282]: `-i {osencoded}`?

**Review by uartie — CHANGES_REQUESTED** [review]

File: lib/ffmpeg/encoderbase.py (hunk 5 above)
Comment (tong1wu) [reply lib/ffmpeg/encoderbase.py]: modified them all, thanks

**Pull request merged**